tests/functional/ingestion/IngestionReader.js

```diff
@@ -101,9 +101,16 @@ describe('ingestion reader tests with mock', function fD() {
 
     before(done => {
         testConfig.s3.port = testPort;
+        const util = require('util');
+        console.log(`\n\n------- mongo config: ${util.inspect(testConfig.queuePopulator.mongo, false, null)}\n\n`);
+        let mongoAuth = '';
+        if (process.env.MONGO_AUTH_PASSWORD && process.env.MONGO_AUTH_USERNAME) {
+            mongoAuth = `${process.env.MONGO_AUTH_USERNAME}:${process.env.MONGO_AUTH_PASSWORD}@`;
+        }
         const mongoUrl =
-            `mongodb://${testConfig.queuePopulator.mongo.replicaSetHosts}` +
-            '/db?replicaSet=rs0';
+            `mongodb://${mongoAuth}${testConfig.queuePopulator.mongo.replicaSetHosts}` +
```

Comment:
I don't know if this would work, the `/db` is the database to authenticate against, which would usually be `admin` unless you explicitly created users in a different way.